### PR TITLE
Fix ist slaves

### PR DESCRIPTION
**Pull request by gabemontero**
@bparees see https://github.com/openshift/origin/pull/14756#issuecomment-309869973

also i inadvertently cut a new release before merging this change .... i'll have to cut a 0.1.22 after 0.1.21 is complete

### CODE REVIEW

File: src/main/java/io/fabric8/jenkins/openshiftsync/ImageStreamWatcher.java [hunk 1]
```diff
@@ -189,7 +189,8 @@ private synchronized void onInitialImageStream(ImageStreamList imageStreams) {
             }
             // for IST, can't set labels, so check annotations
             if (ist != null && hasSlaveLabelOrAnnotation(ist.getMetadata().getAnnotations())) {
-                results.add(this.podTemplateFromData(ist.getMetadata().getName(),
+                // note, pod names cannot have colons
+                results.add(this.podTemplateFromData(ist.getMetadata().getName().replaceAll(":", "."),
```
Comment (bparees) [inline src/main/java/io/fabric8/jenkins/openshiftsync/ImageStreamWatcher.java]: so now "my.imagestream:foo" and "my:imagestream.foo" are going to get mapped to the same thing.  are dots allowed in imagestream names and tag names? i suspect they are.

unfortunately this is hard problem for any separator unless there's a value that's valid in pod names and not valid in imagestreams or tag names.

**Comment by gabemontero** [comment]
what was valid for pod names seemed pretty limited .... lower case
aphanumeric, hyphens, and periods ... that was it
according to the validation error message that came back

**Comment by bparees** [comment]
> what was valid for pod names seemed pretty limited .... lower case aphanumeric, hyphens, and periods ... that was it according to the validation error message that came back

yeah most of this stuff accepts anything that's valid for a DNS name.

there are options to pick a more distinctive separator that's less likely to collide, but i admit even my scenario is relatively contrived.

**Comment by gabemontero** [comment]
this commit passed tests in openshift repo ... given where the separator discussion landed with @bparees going to merge

the jenkins ci jobs are failing with some maven complaint regarding our pom's use of jdk 1.8 when the dependency of the k8s plugin was pulled in by Ryan, though `mvn clean package` works as does `mvn release:prepare release:perform`

will circle back to see if the pom.xml can be tweaked to clean those up after we cycle through last minute 3.6 bugs

**Pull request merged**